Container mnt

mslacken opened the pull request:
## Add standard mounts on `wwctl container exec`

On execution of `wwctl container exec` a configurable number  of mount points will be  available in the chroot. Also the the copy operation of the file `resolv.conf` is on container import is removed, which means that the container is [not] modified on import any more.

mslacken commented:
close #494

anderbubble commented:
@mslacken since I see you configured this in `warewulf.conf`, is `mounts.conf` extraneous?

And just to confirm: this affects container build, too, not just exec and shell?

mslacken commented:
@anderbubble  I introduced `mounts.conf` during the development, but removed it in favor for `warewulf.conf`. This keeps the configuration more simple.
These bind mounts are only used for `exec` and `shell` and will not show up in `build`.
I also added the `node` option, which will give you a best effort view of how a container would look like on the node. @griznog requested something like this.

anderbubble reviewed (changes requested):
Honestly, this is big enough and in parts of the code that I haven't touched enough that I'm not going to be able to give it a very good code review. I did note that `mounts.conf` is still present in the PR, which I think should be removed.

But other than that I just ran it, but there appears to be a problem.

```
[janderson@rocky warewulf]$ sudo wwctl container import docker://docker.io/anderbubble/warewulf-rockylinux:8 anderbubble-rockylinux-8
Copying blob 5461c86c2e54 done  
Copying blob 2ef2dfdac739 done  
Copying blob e4aa6f711855 done  
Copying blob b8fd1f017bbd done  
Copying blob 9231c4f09ac0 done  
Copying blob 6c1511b659e1 done  
Copying blob 154247c0cba1 done  
Copying config e776ae96de done  
Writing manifest to image destination
Storing signatures
2023/02/02 13:10:35  info unpack layer: sha256:5461c86c2e548f89cd5d0c326aaf624b5af719ff4b3a5362f1c8c1f30fea5a1d
2023/02/02 13:10:36  info unpack layer: sha256:2ef2dfdac739d58e532b819c29487535bf6cbfd6f850a82ec6fe911975c3dc08
2023/02/02 13:10:41  info unpack layer: sha256:e4aa6f71185580712bf6110e08622228c7d84e4bad13ce7aff9d0a132108f88e
2023/02/02 13:10:41  info unpack layer: sha256:b8fd1f017bbd97acbb6c77bb59bf92c8d7c1048dd166a26d9f04c5538c83b7d3
2023/02/02 13:10:41  info unpack layer: sha256:9231c4f09ac0e58207e33b7b46bf2706fdde7fa60315ad2718306acdff540e14
2023/02/02 13:10:41  info unpack layer: sha256:6c1511b659e1fb9b3e3223a9db3fcd135dc2acbbf69b4ccb524be63c36511922
2023/02/02 13:10:41  info unpack layer: sha256:154247c0cba17ac52d9a5ffebb7a1291b40a0a8342410ca9f17de24d2335abb7
Updating the container's /etc/resolv.conf
WARN   : user: systemd-resolve:193:193 not present on host
uid/gid not synced, run 
wwctl container syncuser --write anderbubble-rockylinux-8
to synchronize uid/gids.
Building container: anderbubble-rockylinux-8
Created image for VNFS container anderbubble-rockylinux-8: /var/lib/warewulf/container/anderbubble-rockylinux-8.img
Compressed image for VNFS container anderbubble-rockylinux-8: /var/lib/warewulf/container/anderbubble-rockylinux-8.img.gz

[janderson@rocky warewulf]$ sudo wwctl container shell anderbubble-rockylinux-8
ERROR  : exec format error
```

anderbubble commented on Makefile:
```diff
@@ -67,7 +67,7 @@ endif
 
 # Warewulf directory paths
-VARLIST += WWCLIENTDIR WWCONFIGDIR WWPROVISIONDIR WWOVERLAYDIR WWCHROOTDIR WWTFTPDIR WWDOCDIR WWDATADIR
+VARLIST += WWCLIENTDIR WWCONFIGDIR WWPROVISIONDIR WWOVERLAYDIR WWCHROOTDIR WWTFTPDIR WWDOCDIR WWDATADIR TMPDIR
```
I think `TMPDIR` should be renamed so it isn't confused with the canonical environment variable and what is returned by `os.TempDir`.

mslacken commented:
Right its much better to sue the canonical os.TempDir instead of adding a TEMPDIR
The `exec` error is strange, but could be related to your used SHELL, so I added some testing, if $SHELL is used.
